mongoOperator/helpers/MongoMonitoring.py

```diff
@@ -0,0 +1,119 @@
+# Copyright (c) 2018 Ultimaker
+# !/usr/bin/env python
+# -*- coding: utf-8 -*-
+from typing import Callable
+
+from pymongo import monitoring
+from mongoOperator.models.V1MongoClusterConfiguration import V1MongoClusterConfiguration
+
+import logging
+
+
+class CommandLogger(monitoring.CommandListener):
+
+    def started(self, event):
+        logging.debug("Command {0.command_name} with request id "
+                      "{0.request_id} started on server "
+                      "{0.connection_id}".format(event))
+
+    def succeeded(self, event):
+        logging.debug("Command {0.command_name} with request id "
+                      "{0.request_id} on server {0.connection_id} "
+                      "succeeded in {0.duration_micros} "
+                      "microseconds".format(event))
+
+    def failed(self, event):
+        logging.debug("Command {0.command_name} with request id "
+                      "{0.request_id} on server {0.connection_id} "
+                      "failed in {0.duration_micros} "
+                      "microseconds".format(event))
+
+
+class ServerLogger(monitoring.ServerListener):
+
+    def opened(self, event):
+        logging.debug("Server {0.server_address} added to topology "
+                      "{0.topology_id}".format(event))
+
+    def description_changed(self, event):
+        previous_server_type = event.previous_description.server_type
+        new_server_type = event.new_description.server_type
+        if new_server_type != previous_server_type:
+            # server_type_name was added in PyMongo 3.4
+            logging.debug(
+                "Server {0.server_address} changed type from "
+                "{0.previous_description.server_type_name} to "
+                "{0.new_description.server_type_name}".format(event))
+
+    def closed(self, event):
+        logging.debug("Server {0.server_address} removed from topology "
+                      "{0.topology_id}".format(event))
+
+
+class HeartbeatLogger(monitoring.ServerHeartbeatListener):
```

Comment:
Our codestyle states that classes should be in their own file. We should create 4 files for these then. Alternatively we make a single monitoring class that can handle all the different event types.